haskell/tools/ghc_wrapper.py

```diff
@@ -131,13 +131,12 @@ def recompute_abi_hash(ghc, abi_out):
     """Call ghc on the hi file and write the ABI hash to abi_out."""
     hi_file = abi_out.with_suffix("")
 
-    cmd = [ghc, "--show-iface", hi_file]
-    for line in subprocess.check_output(cmd, text=True).splitlines():
-        if "ABI hash:" in line:
-            hash = line.split(":", 1)[1]
-            with open(abi_out, "w") as outfile:
-                print(hash, file=outfile)
-            return
+    cmd = [ghc, "--show-iface-abi-hash", hi_file]
+    for line in subprocess.check_output(cmd, text=True):
+        hash = line[0]
+        with open(abi_out, "w") as outfile:
+            print(hash, file=outfile)
+        return
```

Comment:
This only extracts the first character of the first line.

```suggestion
    return subprocess.check_output(cmd, text=True).split(maxsplit=1)[0]
```

Reply:
Also, when testing that on my machine, I get diagnostic output mangled with the ABI hash:
```console
$ /nix/store/4kafspbkhbf590yd41m8d4m77j06j788-ghc-9.8.2-with-packages/bin/ghc --show-iface-abi-hash buck-out/v2/gen/root/904931f735703749/backend/__backend_infra__/mod-static/src/App.hi | bat
───────┬─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ STDIN
───────┼─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ Loaded package environment from /home/claudio/.ghc/x86_64-linux-9.8.2/environments/default
   2   │ 39589f7ccebb28f6f5b2130cfe55b660
   3   │ 
───────┴─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

```
Using `-v0` suppresses that.

Reply:
ah. 🤦 thanks for the catch!

Reply:
and yeah. GHC should have a clear separation of supplementary information/logging and intended output in terms of `stderr` and `stdout`.